go-controller/pkg/node/udn_isolation.go

```diff
@@ -293,52 +295,77 @@ func (m *UDNHostIsolationManager) updateKubeletCgroup() error {
 // If a new cgroup is created, you load the filtering policy for the new cgroup and then add
 // processes to that cgroup. You only have to follow the right sequence to avoid problems.
 func (m *UDNHostIsolationManager) runKubeletRestartTracker(ctx context.Context) (err error) {
-	conn, err := dbus.NewSystemdConnectionContext(ctx)
+
+	conn, err := dbus.Dial("unix:path=/run/systemd/private", dbus.WithContext(ctx))
 	if err != nil {
-		return fmt.Errorf("failed to connect to systemd: %w", err)
+		return err
 	}
+
 	defer func() {
 		if err != nil {
-			conn.Close()
+			if err := conn.Close(); err != nil {
+				klog.Errorf("Error closing dbus connection for UDN isolation: %v", err)
+			}
 		}
 	}()
 
-	err = conn.Subscribe()
+	// Only use EXTERNAL method, and hardcode the uid (not username)
+	// to avoid a username lookup (which requires a dynamically linked
+	// libc)
+	methods := []dbus.Auth{dbus.AuthExternal(strconv.Itoa(os.Getuid()))}
+
+	err = conn.Auth(methods)
 	if err != nil {
-		return fmt.Errorf("failed to subscribe to systemd events: %w", err)
+		return err
 	}
-	// interval is important here as we need to catch the restart state, before it is running again
-	events, errChan := conn.SubscribeUnitsCustom(50*time.Millisecond, 0, func(u1, u2 *dbus.UnitStatus) bool { return *u1 != *u2 },
-		func(s string) bool {
-			return s != "kubelet.service"
-		})
+
+	signalChan := make(chan *dbus.Signal, 10)
+	conn.Signal(signalChan)
+
 	// run until context is cancelled
 	go func() {
-		waitingForActive := false
 		for {
 			select {
 			case <-ctx.Done():
-				conn.Close()
+				if err := conn.Close(); err != nil {
+					klog.Errorf("Error closing dbus connection for UDN isolation: %v", err)
+				}
 				return
-			case event := <-events:
-				for _, status := range event {
-					if status.ActiveState != "active" {
-						waitingForActive = true
-					} else if waitingForActive {
-						klog.Infof("Kubelet was restarted, re-applying UDN host isolation")
-						err = m.updateKubeletCgroup()
-						if err != nil {
-							klog.Errorf("Failed to re-apply UDN host isolation: %v", err)
-						} else {
-							waitingForActive = false
+			case signal := <-signalChan:
+				klog.V(5).Infof("D-Bus event received: %#v", signal)
```

Comment:
I think this may be too much logging?

Reply:
its V5, you still think I should remove it?

Reply:
I wasn't sure what you can debug with this. Unexpected path that is not parsed correctly?

Reply:
right, i thought in the future if its not working, its probably going to be us parsing events incorrectly, so this would help us identify if that was the bug